pkg/qdr/messaging.go

```diff
@@ -0,0 +1,105 @@
+package qdr
```

Comment:
An observation,  might this be better placed in messaging pkg as it does not contain qdr capabilties?

Reply:
The aim with the messaging package was to be a fairly pure interface. In this package we have a particular implementation that can be used to talk through qdr. I agree that strictly speaking it is an amqp implementation and not tied to the router, but the intent of it is use with the router.

Having the interface is mostly just to ease testing. It also helps reduce direct dependencies between components.